emcc.py

```diff
@@ -863,6 +863,16 @@ def need_llvm_debug_info(options):
           # If not = is specified default to 1
           if '=' not in key:
             key += '=1'
+
+          # Special handling of browser version targets. A version -1 means that the specific version
+          # is not supported at all. Replace those with Infinity to make it possible to compare e.g.
+          # #if MIN_FIREFOX_VERSION < 68
+          try:
```

Comment:
I don't follow. Is this making `-1` mean `Infinity`, and since we compare `MIN_X_VERSION < Infinity` we always get `True` - but don't you want `False` here, as it should mean "not supported"?

Regardless, it seems confusing to have either infinity or -1 mean "not supported", as some comparisons might be `<` and others `>`?

How about having `MAX_X_VERSION` which default to `2147483647` (maximum signed int; a number that fits in a JS smi and also a value we will never reach)?

Reply:
Having MAX_X_VERSION fields does not immediately seem useful? I don't think anyone has ever thought "I don't want to support Firefox/Chrome/Safari that is newer than version Z, but I want to be able to successfully run on older versions than Z."

The idea is that if user sets `-s MIN_FIREFOX_VERSION=68`, then Firefox versions >= 68 will work. If user sets `-s MIN_FIREFOX_VERSION=0`, then all Firefox versions (versions >= 0) will work, and if user sets `-s MIN_FIREFOX_VERSION=-1`, then Firefox is not being targeted at all.

The reason for changing -1 to Infinity internally is that asking users to specify `-s MIN_FIREFOX_VERSION=Infinity` (or `-s MIN_FIREFOX_VERSION=SomeReallyHighNumber`) if they do not wish to target Firefox at all feels awkward (though that will work if someone finds that intuitive to use).

Internally in JS library code, one can then use

```js
#if MIN_FIREFOX_VERSION <= 68
   // code that targets Firefox versions <= 68
#endif

#if MIN_IE_VERSION != TARGET_NOT_SUPPORTED
   // code that is enabled if any Internet Explorer version is targeted
#endif

#if MIN_IE_VERSION == TARGET_NOT_SUPPORTED
   // code that should be enabled only if IE is not being targeted at all
#endif
```

With this kind of scheme, we can avoid needing a `-s MAX_FIREFOX_VERSION=0` directive if one did not want to support Firefox at all.

Reply:
Ah, nice, the `TARGET_NOT_SUPPORTED` makes it clear then.